doubly_linked_list/doubly_linked_list.py

```diff
@@ -44,25 +44,118 @@ def __len__(self):
     return self.length
 
   def add_to_head(self, value):
-    pass
+    if not self.head:
+      self.head = ListNode(value)
+      self.tail = self.head
+      self.length = 1
+      return
+    
+    # self.old_head = self.head
+    # self.new_head = ListNode(value)
+    # self.new_head.next = self.old_head
+    # self.old_head.prev = self.new_head
+    # self.head = self.new_head
+
+    self.head.insert_before(value)
+    self.head = self.head.prev
+    self.length += 1
 
   def remove_from_head(self):
-    pass
+    if not self.head and not self.tail:
+      return False
+    if self.length == 1:
+      old_head = self.head
+      self.head = None
+      self.tail = None
+      self.length = 0
+      return old_head.value
+    if self.length == 2:
+      old_head = self.head
+      self.head = self.tail
+      self.length = 1
+      return old_head.value
+    else:
+      self.old_head = self.head
+      self.head = self.head.next
+      self.length -= 1
+      return self.old_head.value
+
```

Comment:
after removing the head; the new item that will become the head needs its node.prev reset to not have a prev